src/DatabaseLogReader.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace SESP;
+
+use DatabaseLogEntry;
+use DatabaseBase;
+use MWTimestamp;
+use Title;
+use User;
+
+class DatabaseLogReader {
+
+	// The parameters for our query.
+	private $query;
+
+	// The current log
+	private $log;
+
+	// Don't run multiple queries if we don't have to
+	private static $titleCache = [];
+
+	// The db connection
+	private $db;
+
+	// The title key
+	private $titlekey;
+
+	// The type of query being performed
+	private $type;
+
+	/**
+	 * Constructor for reading the log.
+	 *
+	 * @param AppFactory $appFactory injected AppFactory
+	 * @param Title $title page
+	 * @param string $type of log (default: approval)
+	 */
+	public function __construct( DatabaseBase $db, Title $title, $type = 'approval' ) {
+		$this->db = $db;
+		$this->titlekey = $title->getDBKey();
```

Comment:
Why not just require the titleKey (please camelCase) if the whole Title object is not needed?